framework/filters/PageCache.php

```diff
@@ -298,6 +301,11 @@ protected function calculateCacheKey()
         if ($this->varyByRoute) {
             $key[] = Yii::$app->requestedRoute;
         }
+        foreach ($this->variations as &$variation) {
+            if ($variation instanceof \Closure) {
+                $variation = call_user_func($variation, $this);
+            }
+        }
```

Comment:
I don't think that we should modify `$this->variations` directly. This will create non-obvious side effect.